rules-base/api/src/main/java/org/windup/rules/apps/condition/regex/StreamRegexMatcher.java

```diff
@@ -0,0 +1,99 @@
+    {
+        this.listener = listener;
+        Matcher jdkMatcher = Pattern.compile(regex, 0).matcher("");
+        jdkMatcher.useTransparentBounds(true);
+        jdkMatcher.useAnchoringBounds(false);
+        init(new OnStreamStandardMatcher(jdkMatcher), new StreamRegexMatcherProcessor(), 0, 2048);
+    }
+
+    @SuppressWarnings("hiding")
+    protected void init(OnStreamMatcher matcher, MatchProcessor matchProcessor, int minimumLengthOfLookBehind,
+                int newNumberOfChars)
+    {
+
+        ModificationFactory modFactory = new ModificationFactory(minimumLengthOfLookBehind, newNumberOfChars);
+        this.factory = new LineColumnAwareModificationFactory(modFactory);
+        this.matchProcessor = matchProcessor;
+        this.matcher = matcher;
+        this.newNumberOfChars = newNumberOfChars;
+    }
+
+    private LineColumnAwareModificationFactory getMatchFactory()
+    {
+        return (LineColumnAwareModificationFactory) this.factory;
+    }
+
+    private class StreamRegexMatcherProcessor extends DoNothingProcessor implements MatchProcessor
+    {
+        @Override
+        public MatchProcessorResult process(StringBuilder characterBuffer, int firstModifiableCharacterInBuffer, MatchResult matchResult)
+        {
+            LineColumnAwareModificationFactory matchFactory = getMatchFactory();
+            // This is the line number of the start of the buffer, not necessarily the start of the match
+            long lineNumber = matchFactory.getCurrentLine();
+            // the start Column may be after the first char in a line, so use the LineColumnAwareModificationFactory to figure out the actual column
+            // number
+            int startColumn = (int) matchFactory.getCurrentColumn() + matchResult.start();
+            boolean calculateStartColumn = false;
+
+            // now we calculate the actual line number based upon the start line number of the buffer and the index of the match in the buffer
+            char lastChar = 0;
+            for (int i = 0; i < matchResult.start(); i++)
+            {
+                char ch = characterBuffer.charAt(i);
+                if (ch == '\r')
+                {
+                    startColumn = 0;
+                    calculateStartColumn = true;
+                    lineNumber++;
+                }
+                else if (ch == '\n')
+                {
+                    if (lastChar != '\r')
+                    {
+                        startColumn = 0;
+                        calculateStartColumn = true;
+                        lineNumber++;
+                    }
+                }
+                else if (calculateStartColumn)
+                {
+                    startColumn++;
+                }
+                lastChar = ch;
+            }
+
+            StreamRegexMatchedEvent event = new StreamRegexMatchedEvent(matchResult.group(), lineNumber, startColumn);
+            listener.regexMatched(event);
```

Comment:
I did think about adding multiple listeners, but I don't see a usecase at the moment. It would be easy to refactor later if needed, IMO.